workbench-core/infrastructure/src/workbenchAssociateStackToExistingAppRegistryApplication.ts

```diff
@@ -31,5 +32,6 @@ export class WorkbenchAssociateStackToExistingAppRegistryApplication extends Con
     );
 
     importedApplication.associateApplicationWithStack(stack);
+    createAppInsightsConfiguration(stack);
```

Comment:
is there any limitation in the number of CfnApplications we can create?

Reply:
any time we associate the App Registry to a stack we will be creating one

Reply:
Application Insights creates one CloudFormation stack for each application to manage CloudWatch agent installation and configuration for all application components. By default, each AWS account can have 2000 stacks.